app/src/test/java/org/squidfish/tuplenback/data/model/LevelRepositoryTests.kt

```diff
@@ -0,0 +1,456 @@
+package org.squidfish.tuplenback.data.model
+
+import android.app.Application
+import androidx.test.core.app.ApplicationProvider
+import kotlin.test.Test
+import kotlinx.coroutines.test.runTest
+import org.junit.After
+import org.junit.Before
+import org.junit.runner.RunWith
+import org.koin.core.context.stopKoin
+import org.mockito.kotlin.doReturn
+import org.mockito.kotlin.spy
+import org.robolectric.RobolectricTestRunner
+import org.squidfish.tuplenback.data.game.levels.LevelRepository
+import org.squidfish.tuplenback.games.Game
+import org.squidfish.tuplenback.games.GameSettings
+import org.squidfish.tuplenback.presentation.navigation.LevelData
+import org.squidfish.tuplenback.utils.LevelDeserializationError
+import org.squidfish.tuplenback.utils.Result
+import org.squidfish.tuplenback.utils.data
+
+@RunWith(RobolectricTestRunner::class)
+class LevelRepositoryTests {
+    private lateinit var realLevelRepository: LevelRepository
+    private lateinit var spiedLevelRepository: LevelRepository
+    private lateinit var context: Application
+
+    val singleGridLevelConfig = """
+        {
+            "gameType": "Grid",
+            "levels": [
+                {
+                    "levelId": 0,
+                    "settings": {
+                        "recallsBack": 2,
+                        "totalRounds": 3,
+                        "millisPerRound": 1000,
+                        "repeatChance": 20
+                    }
+                }
+            ]
+        }
+    """.trimIndent()
+
+    @Before
+    fun createLevelRepository() {
+        this.context = ApplicationProvider.getApplicationContext<Application>()
+        realLevelRepository = LevelRepository(context)
+        spiedLevelRepository = spy(realLevelRepository)
+    }
+
+    @After
+    fun tearDown() {
+        stopKoin()
+    }
+
+    @Test
+    fun `load level`() = runTest {
+        // Given
+        doReturn(Result.Success(singleGridLevelConfig)).`when`(spiedLevelRepository).readLevelData(Game.Grid)
+
+        // When
+        val settings = spiedLevelRepository.get(LevelData(Game.Grid, 0))
+
+        // Then
+        assert(
+            settings.data == GameSettings(
+                recallsBack = 2,
+                totalRounds = 3,
+                millisPerRound = 1000,
+                repeatChance = 20,
+            ),
```

Comment:
Not that the trailing comma is that big of a deal, but I think there is an `asert` version that takes a lambda. Might be called `assertTrue` though